qiskit/algorithms/optimizers/umda.py

```diff
@@ -0,0 +1,270 @@
+            vqe = QAOA(opt,
+                       quantum_instance=QuantumInstance(backend=backend),
+                       reps=p)
+
+            result = vqe.compute_minimum_eigenvalue(operator=qubit_op)
+
+        If it is desired to modify the percentage of individuals considered to update the
+        probabilistic model, then this code can be used. Here for example we set the 60% instead
+        of the 50% predefined.
+
+        .. code-block:: python
+
+            opt = UMDA(maxiter=100, size_gen=20, n_variables=p*2, alpha = 0.6, disp=True)
+
+            backend = Aer.get_backend('statevector_simulator')
+            vqe = QAOA(opt,
+                       quantum_instance=QuantumInstance(backend=backend),
+                       reps=p)
+
+            result = vqe.compute_minimum_eigenvalue(operator=qubit_op)
+
+
+    References:
+
+        [1]: Vicente P. Soloviev, Pedro Larrañaga and Concha Bielza (2022, July). Quantum Parametric
+        Circuit Optimization with Estimation of Distribution Algorithms. In 2022 The Genetic and
+        Evolutionary Computation Conference (GECCO). DOI: https://doi.org/10.1145/3520304.3533963
+
+        [2]: Vicente P. Soloviev. Python package EDAspy.
+        https://github.com/VicentePerezSoloviev/EDAspy.
+    """
+
+    best_mae_global = 9999999
+    best_ind_global = 9999999
+    history = []
+    evaluations = 0
+
+    elite_factor = 0.4
+    std_bound = 0.3
+
+    setting = "---"
+
+    def __init__(
+        self, maxiter: int, size_gen: int, n_variables: int, alpha: float = 0.5, disp: bool = False
+    ) -> None:
+        r"""
+        Args:
+            maxiter: Maximum number of function evaluations.
+            size_gen: Population size of each generation.
+            n_variables: Number of variables to be optimized. For example in QAOA the number of variables
+            is 2p, where p is the number of layers of the circuit.
+            alpha: Percentage [0, 1] of the population to be selected as elite selection.
+            disp: Set to True to print convergence messages.
+        """
+
+        self.disp = disp
+        self.size_gen = size_gen
+        self.max_iter = maxiter
+        self.alpha = alpha
+        self.n_variables = n_variables
+        self.vector = self._initialization()
+        self.dead_iter = self.max_iter / 5
+
+        self.best_mae_global = 999999999999
+        self.truncation_length = int(size_gen * alpha)
+
+        # initialization of generation
+        self.generation = algorithm_globals.random.normal(
+            self.vector[0, :], self.vector[1, :], [self.size_gen, self.n_variables]
+        )
+
+        super().__init__(method="UMDA")
+
+    def _initialization(self):
+        vector = np.zeros((4, self.n_variables))
+
+        vector[0, :] = np.pi  # mu
+        vector[1, :] = 0.5  # std
+        vector[2, :] = 0  # min
+        vector[3, :] = np.pi * 2  # max
+
+        return vector
```

Comment:
I had wondered whether a simple dataclass called say probabilty, with fields mu, std, min, max would lead to more readable code overall where its used below etc. But maybe you are constrained here by what can be passed to/used by scipy.stats.norm

Reply:
I have removed lines 183-184 from this piece of code you show me. With this modification, I do not think it is useful to create a new class, as we are only using a matrix of two rows where first are the means and the second the stds. Think the best way is to leave it as it is now.